lib/lvol/lvol.c

```diff
@@ -956,6 +956,51 @@ lvol_create_open_cb(void *cb_arg, struct spdk_blob *blob, int lvolerrno)
 	free(req);
 }
 
+static void
+lvol_create_snapshot_open_cb(void *cb_arg, struct spdk_blob *blob, int lvolerrno)
```

Comment:
I wonder if this function is needed. Can't we have just one our callback which proxies calls to the original callback passed to 'create_lvol_snapshot()' ? Then we just need to check against 'lvolerrno' and reset cache there.

Reply:
Yes I think that would be simpler, and reduce some of this code duplication.

Reply:
Updated the code.  Please check once.